clientcompat issue with V8

Thanks for this project!

I am the author of a [protoc-plugin](https://github.com/timostamm/protobuf-ts) that supports Twirp clients. 

Unfortunately, clientcompat started failing with the V8 release.

```shell
$ go get github.com/twitchtv/twirp/clientcompat
$ clientcompat -client clientcompat/client-runner.js
# some lines omitted, all PASS
Testing empty value... FAIL: client has wrong response, have  want 
Testing request value formatting... FAIL: client has wrong response, have v:1 want v:1
Testing handling invalid error formatting from server... PASS
FAILED with 2 failures, 19 successes
```

Exactly the same client passes with `go get github.com/twitchtv/twirp/clientcompat@v7.0.0`. 

Between v7 and v8, my client is receiving the same request payload via stdin and is writing the same server response to stdout. This is true for both failed tests.

Any idea where this could be coming from?

`clientcompat` is used as described in your [README](https://github.com/twitchtv/twirp/blob/master/clientcompat/README.md). A [make target](https://github.com/timostamm/protobuf-ts/blob/afce26fc55a9aed773c6201aeb66ae76a10223ab/packages/twirp-transport/Makefile#L36-L44) runs (as part of CI and release build):

- compiles your clientcompat.proto using our plugin
- compiles [our test client](https://github.com/timostamm/protobuf-ts/blob/afce26fc55a9aed773c6201aeb66ae76a10223ab/packages/twirp-transport/clientcompat/client.ts), which works just like your [gocompat.go](https://github.com/twitchtv/twirp/blob/fc964007b6a7badaa284114f13ebe08ce821354a/clientcompat/gocompat/gocompat.go) example implementation
- `go get`s your `clientcompat`
- runs your `clientcompat` against our test client

BTW, conformance testers like clientcompat are super helpful to implement against!

Yes, your CI does a `go test` but does not include your make [target for clientcompat](https://github.com/twitchtv/twirp/blob/fc964007b6a7badaa284114f13ebe08ce821354a/Makefile#L26). If you run that manually you get a failure:

```shell
➜  twirp git:(master) ✗ make test_go_client
# Recompile and install generator
GOBIN="$PWD/bin" go install -v ./protoc-gen-twirp
GOBIN="$PWD/bin" go install -v ./protoc-gen-twirp_python
# Generate code from go:generate comments
go generate ./...
./build/clientcompat -client ./build/gocompat
Testing noop without error... PASS
Testing "canceled" error parsing... PASS
Testing "unknown" error parsing... PASS
Testing "invalid_argument" error parsing... PASS
Testing "deadline_exceeded" error parsing... PASS
Testing "not_found" error parsing... PASS
Testing "bad_route" error parsing... PASS
Testing "already_exists" error parsing... PASS
Testing "permission_denied" error parsing... PASS
Testing "unauthenticated" error parsing... PASS
Testing "resource_exhausted" error parsing... PASS
Testing "failed_precondition" error parsing... PASS
Testing "aborted" error parsing... PASS
Testing "out_of_range" error parsing... PASS
Testing "unimplemented" error parsing... PASS
Testing "internal" error parsing... PASS
Testing "unavailable" error parsing... PASS
Testing "data_loss" error parsing... PASS
Testing empty value... FAIL: client has wrong response, have  want 
Testing request value formatting... FAIL: client has wrong response, have v:1 want v:1
Testing handling invalid error formatting from server... PASS
FAILED with 2 failures, 19 successes
make: *** [test_go_client] Error 1
```

--- a/clientcompat/main.go
+++ b/clientcompat/main.go
@@ -17,11 +17,11 @@ import (
 	"context"
 	"flag"
 	"fmt"
+	"google.golang.org/protobuf/proto"
 	"log"
 	"net/http"
 	"net/http/httptest"
 	"os"
-	"reflect"
 
 	"github.com/twitchtv/twirp"
 	"github.com/twitchtv/twirp/clientcompat/internal/clientcompat"
@@ -161,7 +161,7 @@ func testMethod(cc *clientCompat, s *httptest.Server, clientBin string) {
 			return
 		}
 
-		if !reflect.DeepEqual(resp, wantResp) {
+		if !proto.Equal(resp, wantResp) {
 			fail("client has wrong response, have %+v want %+v", resp, wantResp)
 			return
 		}

